src/cplscheme/CouplingData.hpp

```diff
@@ -25,44 +46,52 @@ struct CouplingData { // @todo: should be a class from a design standpoint. See
     return data->values();
   }
 
-  /// Data values of previous iteration (1st col) and previous time windows.
-  DataMatrix oldValues;
+  void storeIteration()
+  {
+    _lastIteration = this->values();
+  }
 
-  mesh::PtrData data;
+  const Eigen::VectorXd lastIteration() const
+  {
+    return _lastIteration;
+  }
 
-  mesh::PtrMesh mesh;
+  [[deprecated("lastIteration should be read only!")]] void writeLastIteration(Eigen::VectorXd value)
+  {
+    _lastIteration = value;
+  }
 
-  ///  True, if the data values if this CouplingData requires to be initialized by a participant.
-  bool requiresInitialization;
+  int getMeshID()
+  {
+    return mesh->getID();
+  }
 
-  int getDimensions()
+  std::vector<int> getVertexOffsets()
   {
-    PRECICE_ASSERT(data != nullptr);
-    return data->getDimensions();
+    return mesh->getVertexOffsets();
   }
 
+  ///  True, if the data values if this CouplingData requires to be initialized by a participant.
```

Comment:
```suggestion
  ///  True, if the data values of this CouplingData require to be initialized by a participant.
```
?
"a" participant?

Reply:
Should be "this participant". Changed it in e26437c63eaf5e7df470df7cfd5a902dc7440d2c.